### PR TITLE
New admin punishment page

### LINKED ISSUE #261: [Feature Request] 管理操作改为页面
目前的管理操作都使用输入框和对话框来进行，容易误触，且不便于在管理时查看帖子内容。可以改成如 Swift 版一样的页面或 Bottom Sheet，如下图。

图片

### PR DESCRIPTION
**Pull request by HydrogenC**
WIP, shall close #261.

### CODE REVIEW

**Comment by w568w** [comment]
Being very busy these days, and as soon as I have time I'll start reviewing the parts of it that have something to do with the admin page.

Are there any more commits unrelated to the admin page (i.e. about tweaking the UI layout)? I'll cherry-pick them.

**Comment by HydrogenC** [comment]
I believe `074ace9`, `a8a27f6`, `3ec1fb5` are fixes that have nothing to do with the admin page.

**Comment by HydrogenC** [comment]
Is this PR superseded by #318 already?

**Comment by HydrogenC** [comment]
I accidentally did the same thing. I gotta have `f80a9cb` reverted.

**Comment by w568w** [comment]
Has it been already done, or still working in progress?

**Comment by HydrogenC** [comment]
> Sorry for too late reply. Has it been already done, or still working in progress?

Feature complete already. But I haven't done any code style-checking yet. I will do that asap.

**Comment by w568w** [comment]
> Tested to work on PC, but not tested on mobile platforms yet (help wanted). Since this PR replaces the old solution, it has to be well-tested (however testing is not possible now since the server is down).

That's fine. I may merge this after the FDUHole server goes up again... As for now, I suggest keeping the PR open.

**Comment by w568w** [comment]
I will test it on Android with the development server soon.

File: lib/page/opentreehole/admin_operation.dart [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+
+  if (result == null) return false;
+
+  try {
+    final operation = result! as AdminOperationInfo;
+    int? response;
+    if (operation.isDelete) {
+      response = await OpenTreeHoleRepository.getInstance().adminDeleteFloor(
+          operation.floorId,
+          operation.reason.isEmpty ? "未知原因" : operation.reason);
```
Comment (w568w) [inline lib/page/opentreehole/admin_operation.dart:48]: You don't need to hard-code a default reason here: the backend has a default reason string when receiving an empty string, which is usually "违反社区规范". 

Anyway, this default string should be decided and maintained by the backend, not by us.

**Review by w568w — CHANGES_REQUESTED** [review]
Mostly LGTM. Except for a minor issue.

**Review by w568w — APPROVED** [review]

**Pull request merged**